### PR TITLE
Add more information to README

### PR DESCRIPTION
**Pull request by ShanBai6**
Please review all sections. I am sure there will be missing parts to this.

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -1 +1,67 @@
...
+# Presto Manager
+
+A tool for managing Presto clusters.
+
+## Getting Started
+
+These instructions will get you a copy of the project up and running on your local machine for development and testing purposes.
+
+### Prerequisites
+
```
Comment (jirassimok) [inline README]: Make this a bulleted list (put each item on a line starting with `- ` (dash, space)).

File: README [hunk 2]
```diff
@@ -1 +1,67 @@
...
+Java 8, available at (http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html)
+Maven 3+, available at (https://maven.apache.org/download.cgi)
```
Comment (jirassimok) [inline README]: Maven is not required, because of the Maven Wrapper.

File: README [hunk 3]
```diff
@@ -1 +1,67 @@
...
+Python 2.4+, available at (https://www.python.org/download/releases/2.4/)
```
Comment (jirassimok) [inline README]: I would say "available at python.org", and link the general downloads page.

```markdown
- Python 2.4+, available at [python.org](https://www.python.org/downloads/)
```
Which displays as
- Python 2.4+, available at [python.org](https://www.python.org/downloads/)

File: README [hunk 4]
```diff
@@ -1 +1,67 @@
...
+
+### Installing
+
+Clone the full distribution from (https://github.com/prestodb/presto-manager)
```
Comment (jirassimok) [inline README]: "Clone the git repository:"
No need to give the url here and on the next line.

File: README [hunk 5]
```diff
@@ -1 +1,67 @@
...
+
```
Comment (jirassimok) [inline README]: This line break isn't necessary, and it makes a bit more sense to break only _after_ each step of the instructions (and it makes it look nicer in plain text).

File: README [hunk 6]
```diff
@@ -1 +1,67 @@
...
+```
+git clone https://github.com/prestodb/presto-manager.git
+```
+Go the the directory you cloned the distribution
+
+```
+cd /directory/of/the/clone/presto-manager
```
Comment (jirassimok) [inline README]: `cd presto-manager`; the cloned repo will be in the same folder you are in. Actually, this step probably belongs in "Compiking and Running".

File: README [hunk 7]
```diff
@@ -1 +1,67 @@
...
+```
+Now you should be on the top level of `presto-manager` directory
```
Comment (jirassimok) [inline README]: Not necessary.

File: README [hunk 8]
```diff
@@ -1 +1,67 @@
...
+
+
+## Compiling and starting
```
Comment (jirassimok) [inline README]: "Compiling and Running"

File: README [hunk 9]
```diff
@@ -1 +1,67 @@
...
+
+Use Maven to compile and generate executable JAR for the project
+
+```
+mvn clean install
```
Comment (jirassimok) [inline README]: `./mvnw clean install`

File: README [hunk 10]
```diff
@@ -1 +1,67 @@
...
+
+```
+cd /directory/of/the/clone/presto-manager
+```
+Now you should be on the top level of `presto-manager` directory
+
+
+## Compiling and starting
+
+Use Maven to compile and generate executable JAR for the project
```
Comment (jirassimok) [inline README]: "Use the Maven Wrapper ..."

File: README [hunk 11]
```diff
@@ -1 +1,67 @@
...
+
+```
+mvn clean install
+```
+
+Then `cd` into ./presto-manager-agent/target and ./presto-manager-controller/target, respectively.
```
Comment (jirassimok) [inline README]: Instead of giving instructions to change directory, why not give instructions to use the full path to the executable from the project root?

(And say "change directory" instead of "`cd`").

File: README [hunk 12]
```diff
@@ -1 +1,67 @@
...
+
+Execute
```
Comment (jirassimok) [inline README]: I'd make this a header (level 4 or 5).

File: README [hunk 13]
```diff
@@ -1 +1,67 @@
...
+
+```
+java -jar AgentServer.jar
+```
+And
+
+```
+java -jar ControllerServer.jar
+```
+These command would start the server for Agent and Controller.
```
Comment (jirassimok) [inline README]: "These commands will ..."

File: README [hunk 14]
```diff
@@ -1 +1,67 @@
...
+
+## Usage
+
+Use Postman, a Google chrome plugin, to send Http Request to the Controller Server.
```
Comment (jirassimok) [inline README]: Any http client works, as long as it can send the right type of requests.

File: README [hunk 15]
```diff
@@ -1 +1,67 @@
...
+
+Example would be
+
+```
+GET http://localhost:8080/v1/config?scope=cluster
+```
+which would get names of configuration files back to user.
```
Comment (jirassimok) [inline README]: `get` → `send`

File: README [hunk 16]
```diff
@@ -1 +1,67 @@
-Presto Manager: A tool for managing Presto clusters.
+# Presto Manager
+
+A tool for managing Presto clusters.
+
+## Getting Started
+
+These instructions will get you a copy of the project up and running on your local machine for development and testing purposes.
```
Comment (sudarsanm94) [inline README]: instance of the project

File: README [hunk 17]
```diff
@@ -1 +1,67 @@
...
+```
+java -jar ControllerServer.jar
+```
+These command would start the server for Agent and Controller.
+
+## Usage
+
+Use Postman, a Google chrome plugin, to send Http Request to the Controller Server.
+
+Example would be
```
Comment (sudarsanm94) [inline README]: An Example would be

File: README (hunk 14 above)
Comment (sudarsanm94) [inline README]: +1

File: README (hunk 13 above)
Comment (sudarsanm94) [inline README]: just `start  Agent and the Controller` would be good.

File: README [hunk 18]
```diff
@@ -1 +1,67 @@
...
+
+```
+GET http://localhost:8080/v1/config?scope=cluster
+```
```
Comment (sudarsanm94) [inline README]: May be add a note saying that 
use `/v1/presto/status/scope=ALL` to get the node ID and they can use the specific node IDs for future use.

File: README (hunk 16 above)
Comment (ShanBai6) [inline README]: Fixed.

File: README [hunk 19]
```diff
@@ -1 +1,67 @@
-Presto Manager: A tool for managing Presto clusters.
+# Presto Manager
+
+A tool for managing Presto clusters.
+
+## Getting Started
+
+These instructions will get you a copy of the project up and running on your local machine for development and testing purposes.
+
+### Prerequisites
```
Comment (ShanBai6) [inline README]: Updated. Added OS requirement in getting started.

File: README [hunk 20]
```diff
@@ -1 +1,67 @@
...
+
+Java 8, available at (http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html)
```
Comment (ShanBai6) [inline README]: I will give both. In case they run it in docker

Comment (ShanBai6) [inline README]: And maybe develop it in windows/mac

File: README (hunk 3 above)
Comment (ShanBai6) [inline README]: Updated

File: README (hunk 2 above)
Comment (ShanBai6) [inline README]: Deleted

File: README (hunk 4 above)
Comment (ShanBai6) [inline README]: Deleted

File: README (hunk 9 above)
Comment (ShanBai6) [inline README]: Fixed

File: README (hunk 10 above)
Comment (ShanBai6) [inline README]: Fixed

File: README (hunk 18 above)
Comment (ShanBai6) [inline README]: All responses would return the node id

File: README [hunk 21]
```diff
@@ -1 +1,62 @@
...
+A tool for managing Presto clusters. Targeted for Linux OS, not guarentee to work on Windows/MacOS.
+
+## Getting Started
+
+These instructions will get you a instance of the project up and running on your local machine for development and testing purposes.
+
+### Prerequisites
+
+- JDK 8, available at (http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html)
+         Alternative  (http://openjdk.java.net/install/)
```
Comment (jirassimok) [inline README]: To improve readability in the plaintext, the links can be placed in link references.

```markdown
- Java Development Kit 8, available at [oracle.com][Oracle Java] or [openjdk.java.net][OpenJDK]

[OpenJDK]: http://openjdk.java.net/install
[Oracle Java]: http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html
```
(The last two lines shoud be near the bottom of the document.)

Though maybe [this page](https://docs.oracle.com/javase/8/docs/technotes/guides/install/install_overview.html) would be better as the Oracle link.

File: README [hunk 22]
```diff
@@ -1 +1,62 @@
-Presto Manager: A tool for managing Presto clusters.
+# Presto Manager
+
+A tool for managing Presto clusters. Targeted for Linux OS, not guarentee to work on Windows/MacOS.
```
Comment (jirassimok) [inline README]: The intended operating system should probably be listed as a prerequisite.

File: README [hunk 23]
```diff
@@ -1 +1,62 @@
...
+### Installing
+
+Clone the full distribution
+```
+git clone https://github.com/prestodb/presto-manager.git
+```
+
+## Compiling and Running
+
+Go the the directory you cloned the distribution
```
Comment (jirassimok) [inline README]: "the project directory"

File: README [hunk 24]
```diff
@@ -1 +1,62 @@
...
+
+```
+java -jar ./presto-manager-agent/target/AgentServer.jar
+```
+And
+
+```
+java -jar ./presto-manager-controller/target/ControllerServer.jar
+```
+These command will start the Agent and the Controller.
```
Comment (jirassimok) [inline README]: Put this before you list the comments.
````markdown
These commands will start the Agent and Controller, respectively:
```
java -jar ./presto-manager-agent/target/AgentServer.jar
java -jar ./presto-manager-controller/target/ControllerServer.jar
```
````

File: README [hunk 25]
```diff
@@ -1 +1,62 @@
...
+
+## Usage
+
+Any Http client will work as long as it could send basic types of Http requests, i.e GET, PUT, POST, DELETE, etc.
+Postman, a Google chrome plugin, is recommended for its friendly user interface.
+
+An example would be
```
Comment (jirassimok) [inline README]: "For example, this request will receive a list of available configuration files as its response."

File: README (hunk 22 above)
Comment (sudarsanm94) [inline README]: Yes. Not guaranteed to work... doesn't look good

File: README [hunk 26]
```diff
@@ -1 +1,62 @@
-Presto Manager: A tool for managing Presto clusters.
+# Presto Manager
+
+A tool for managing Presto clusters. Targeted for Linux OS, not guarentee to work on Windows/MacOS.
+
+## Getting Started
+
+These instructions will get you a instance of the project up and running on your local machine for development and testing purposes.
```
Comment (sudarsanm94) [inline README]: `an`

File: README (hunk 23 above)
Comment (sudarsanm94) [inline README]: typo. Go `to` the

File: README (hunk 24 above)
Comment (sudarsanm94) [inline README]: The following commands will start the Agent and the Controller, respectively

File: README [hunk 27]
```diff
@@ -1 +1,62 @@
...
+And
+
+```
+java -jar ./presto-manager-controller/target/ControllerServer.jar
+```
+These command will start the Agent and the Controller.
+
+## Usage
+
+Any Http client will work as long as it could send basic types of Http requests, i.e GET, PUT, POST, DELETE, etc.
```
Comment (sudarsanm94) [inline README]: as long as it could send basic Http requests

File: README (hunk 25 above)
Comment (sudarsanm94) [inline README]: this request will `retrieve` a list of

File: README (hunk 18 above)
Comment (sudarsanm94) [inline README]: Ok. Then you can state some way to send request to a particular node.

File: README.md [hunk 28]
```diff
@@ -0,0 +1,63 @@
...
+java -jar ./presto-manager-controller/target/ControllerServer.jar
+```
+
+## Usage
+
+Any Http client will work as long as it could send basic Http requests
+i.e GET, PUT, POST, DELETE, etc.
+Postman, a Google chrome plugin, is recommended for its friendly user interface.
+
+"For example, this request will receive a list of available configuration files as its response."
```
Comment (jirassimok) [inline README.md]: (Take out the quotes.)

File: README.md [hunk 29]
```diff
@@ -0,0 +1,63 @@
...
+```
+GET http://localhost:8080/v1/config?scope=cluster
+```
+
+Please refer to the swagger API documentation for more information.
+
+## License
+
+This project is licensed under the Apache License
```
Comment (jirassimok) [inline README.md]: End this line with some form of punctuation; it renders on the same line as the next line.

File: README.md [hunk 30]
```diff
@@ -0,0 +1,63 @@
...
+
+These commands will start the Agent and Controller, respectively:
+```
+java -jar ./presto-manager-agent/target/AgentServer.jar
+java -jar ./presto-manager-controller/target/ControllerServer.jar
+```
+
+## Usage
+
+Any Http client will work as long as it could send basic Http requests
```
Comment (jirassimok) [inline README.md]: `could` → `can`

File: README.md [hunk 31]
```diff
@@ -0,0 +1,63 @@
...
+i.e GET, PUT, POST, DELETE, etc.
```
Comment (jirassimok) [inline README.md]: This line should have some formatting, but I'm not sure what kind.

Comment (ShanBai6) [inline README.md]: I forced that to be two lines.

File: README.md (hunk 30 above)
Comment (ShanBai6) [inline README.md]: updated

File: README.md (hunk 29 above)
Comment (ShanBai6) [inline README.md]: Added

File: README.md (hunk 28 above)
Comment (ShanBai6) [inline README.md]: Done

File: README.md [hunk 32]
```diff
@@ -0,0 +1,63 @@
...
+### Installing
+
+Clone the full distribution
+```
+git clone https://github.com/prestodb/presto-manager.git
+```
+
+## Compiling and Running
+
+Go to the project directory you cloned the distribution
```
Comment (sudarsanm94) [inline README.md]: Change to the project directory where you cloned the distribution

File: README (hunk 18 above)
Comment (sudarsanm94) [inline README]: address this comment

**Comment by sudarsanm94** [comment]
Address two of my comments and it is good to go.

File: README (hunk 18 above)
Comment (ShanBai6) [inline README]: I will add it in swagger

File: README.md [hunk 33]
```diff
@@ -0,0 +1,63 @@
...
+### Installing
+
+Clone the full distribution
+```
+git clone https://github.com/prestodb/presto-manager.git
+```
+
+## Compiling and Running
+
+Change to the project directory you cloned the distribution
```
Comment (sudarsanm94) [inline README.md]: `where` you cloned the distribution

File: presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/AbstractControllerAPI.java [hunk 34]
```diff
@@ -0,0 +1,53 @@
...
+ */
+package com.teradata.prestomanager.controller;
+
+import com.teradata.prestomanager.common.ApiRequester;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.core.Response;
+
+import java.util.Collection;
+import java.util.UUID;
```
Comment (sudarsanm94) [inline presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/AbstractControllerAPI.java]: Unrelated file

File: README.md (hunk 33 above)
Comment (ShanBai6) [inline README.md]: Not needed
